test/csuite/wt8246_compact_rts_data_correctness/main.c

```diff
@@ -0,0 +1,375 @@
+static char value_c[] = "CC";
+static char value_d[] = "DD";
+
+/* Structures definition. */
+struct thread_data {
+    WT_CONNECTION *conn;
+    const char *uri;
+    uint32_t begin, end;
+    bool update;
+};
+
+static void sig_handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/* Forward declarations. */
+static int run_test(bool column_store, const char *uri);
+static void populate(WT_SESSION *session, const char *uri);
+static void workload_compact(const char *, const char *, const char *uri);
+static void remove_records(WT_SESSION *session, const char *uri, int commit_ts);
+static void run_compact(WT_SESSION *session, const char *uri);
+static void large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts);
+static void check(WT_SESSION *session, const char *uri, char *value, int commit_ts);
+
+/*
+ * Signal handler to catch if the child died unexpectedly.
+ */
+static void
+sig_handler(int sig)
+{
+    pid_t pid;
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /*
+     * The core file will indicate why the child exited. Choose EINVAL here.
+     */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+/* Methods implementation. */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+
+    testutil_assert(run_test(false, opts->uri) == EXIT_SUCCESS);
+
+    testutil_assert(run_test(true, opts->uri) == EXIT_SUCCESS);
+
+    testutil_cleanup(opts);
+
+    return (EXIT_SUCCESS);
+}
+
+static void
+run_compact(WT_SESSION *session, const char *uri)
+{
+    printf("Running compact...\n");
+    /* Perform compact operation. */
+    testutil_check(session->compact(session, uri, NULL));
+}
+
+static int
+run_test(bool column_store, const char *uri)
+{
+    struct sigaction sa;
+    struct stat sb;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    pid_t pid;
+    uint64_t oldest_ts, stable_ts;
+    int status;
+    char ckpt_file[2048];
+    char home[1024];
+    char ts_string[WT_TS_HEX_STRING_SIZE];
+
+    testutil_work_dir_from_path(
+      home, sizeof(home), column_store ? working_dir_col : working_dir_row);
+
+    printf("Work directory: %s\n", home);
+    testutil_make_work_dir(home);
+
+    /* Fork a child to create tables and perform operations on them. */
+    memset(&sa, 0, sizeof(sa));
+    sa.sa_handler = sig_handler;
+    testutil_checksys(sigaction(SIGCHLD, &sa, NULL));
+    testutil_checksys((pid = fork()) < 0);
+
+    if (pid == 0) { /* child */
+
+        workload_compact(home, column_store ? table_config_col : table_config_row, uri);
+
+        /*
+         * We do not expect test to reach here. The child process should have been killed by the
+         * parent process.
+         */
+        printf("Child finished processing...\n");
+        return (EXIT_FAILURE);
+    }
+
+    sprintf(ckpt_file, compact_file_fmt, home);
```

Comment:
It's better to use `__wt_snprintf` here. There's another usage in `workload_compact`.

Reply:
Fixed